### 파이썬 DB 연동
- CRUD 작업 처리
- CREATE / READ / UPDATE / DELETE
- INSERT / SELECT / UPDATE / DELETE 

### 파이썬에서 MySQL 연동 패키지 지원
- pymysql 패키지 사용 

In [1]:
# pymysql 패키지 설치 
!pip install pymysql

In [2]:
import pymysql

#### DB 연동 순서
- (1) DB 연결   
- (2) cursor 객체 생성 
- (3) sql 구문 실행 
- (4) sql 실행 결과 추출 (출력)

In [3]:
# 1. DB 연결 
# 호스트명, 포트번호, 로그인 계정, 비밀번호, CHARSET 파라미터로 지정 
conn = pymysql.connect(host='localhost', 
                       port=3306, 
                       user='root', 
                       passwd='1234',
                       charset='utf8')
# 연결 객체 반환 

In [4]:
# 2. cursor 객체 생성 
# cursor 객체를 통해서 메소드 호출 
cursor = conn.cursor()

In [5]:
# 3. sql 문 실행 
# 데이터베이스 확인 sql 문 실행 
# 실행 : execute() 메소드 
sql = 'show databases'
cursor.execute(sql)

# 실행 결과 : 데이터베이스 개수 반환 

11

In [6]:
# 4. sql 실행 결과 추출 (출력)
# 서버로부터 받은 결과를 출력 
# fetchall() 메소드 사용 : 모든 데이터를 한꺼번에 클라이언트로 받아올 때 사용 
result = cursor.fetchall()
result

(('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('shopdb',),
 ('sqldb1',),
 ('sqldb2',),
 ('sqldb3',),
 ('sys',),
 ('testdb',),
 ('world',))

In [7]:
# !pip install cryptography

In [8]:
# 새로운 데이터베이스 생성 
sql = 'create database beauty_shop'
cursor.execute(sql)

1

In [9]:
# beauty_shop 데이터베이스 생성 확인
sql = "show databases"
cursor.execute(sql)
result = cursor.fetchall()
result

(('beauty_shop',),
 ('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('shopdb',),
 ('sqldb1',),
 ('sqldb2',),
 ('sqldb3',),
 ('sys',),
 ('testdb',),
 ('world',))

In [10]:
# 데이터베이스 삭제 
sql = 'drop database beauty_shop'
cursor.execute(sql)

0

In [11]:
# beauty_shop 데이터베이스 삭제 확인
sql = "show databases"
cursor.execute(sql)
result = cursor.fetchall()
result

(('information_schema',),
 ('mysql',),
 ('performance_schema',),
 ('sakila',),
 ('shopdb',),
 ('sqldb1',),
 ('sqldb2',),
 ('sqldb3',),
 ('sys',),
 ('testdb',),
 ('world',))

In [12]:
# 사용할 데이터베이스 선택 
sql = 'use sqldb3'
cursor.execute(sql)

0

In [13]:
# 테이블 생성 
sql = """
    create table shop(
        proNo int auto_increment not null primary key,
        prdName varchar(100),
        prdPrice float,
        prdDisPrice float
    )
"""

cursor.execute(sql)

0

In [14]:
# 생성된 테이블 확인 
sql = "show tables"
cursor.execute(sql)
cursor.fetchall()

(('book',),
 ('booksale',),
 ('client',),
 ('client_view',),
 ('flower',),
 ('movie',),
 ('new_book1',),
 ('publisher',),
 ('shop',))

In [15]:
# 테이블 삭제 
sql = 'drop table shop'
cursor.execute(sql)

0

In [16]:
# 테이블 구조 확인
sql = 'desc book'
cursor.execute(sql)
cursor.fetchall()
# 열이름, 데이터타입(크기), null 여부, 키, 디폴트값, 기타

(('bookNo', 'varchar(10)', 'NO', 'PRI', None, ''),
 ('bookName', 'text', 'YES', '', None, ''),
 ('bookAuthor', 'text', 'YES', '', None, ''),
 ('bookPrice', 'int', 'YES', '', None, ''),
 ('bookDate', 'date', 'YES', '', None, ''),
 ('bookStock', 'int', 'YES', '', None, ''),
 ('pubNo', 'int', 'YES', '', None, ''))

In [17]:
###################################################################

In [18]:
# CRUD 작업 처리
# CREATE / READ / UPDATE / DELETE
# INSERT / SELECT / UPDATE / DELETE

In [19]:
# SELECT 

In [20]:
# book 테이블 데이터 조회 
sql = 'select * from book'
cursor.execute(sql)
result = cursor.fetchall()
result

(('1001', '데이터베이스 이론', '홍길동', 22000, datetime.date(2019, 11, 11), 5, 3),
 ('1002', '자바 프로그래밍', '이몽룡', 25000, datetime.date(2021, 12, 12), 4, 1),
 ('1003', '인터넷프로그래밍', '성춘향', 30000, datetime.date(2019, 2, 10), 10, 2),
 ('1004', '안드로이드 프로그래밍', '변학도', 23000, datetime.date(2017, 10, 10), 2, 1),
 ('1005', '안드로이드 앱', '강길동', 26000, datetime.date(2020, 1, 11), 5, 2),
 ('1006',
  'MS SQL SERVER 2014',
  '박지성',
  35000,
  datetime.date(2020, 3, 25),
  7,
  3),
 ('1007', 'HTML & CSS', '손흥민', 18000, datetime.date(2023, 9, 30), 3, 1),
 ('1008', 'MFC 입문', '류현진', 20000, datetime.date(2015, 12, 12), 5, 1),
 ('1009', '안드로이드 게임 제작', '나길동', 33000, datetime.date(2018, 10, 31), 5, 2),
 ('1010', 'C++객체지향 프로그래밍', '추신수', 24000, datetime.date(2019, 4, 20), 2, 3),
 ('1011', 'JSP 웹 프로그래밍', '김연아', 27000, datetime.date(2020, 10, 23), 8, 1),
 ('1012', '해커들의 해킹 기법', '손연재', 32000, datetime.date(2018, 7, 7), 1, 2),
 ('1013', '자료구조', '홍길동', 19000, datetime.date(2020, 1, 20), 4, 1),
 ('1014', '파이썬', '성춘향', 35000, dateti

In [21]:
result[0][0]

'1001'

In [22]:
for re in result:
    print(re)

('1001', '데이터베이스 이론', '홍길동', 22000, datetime.date(2019, 11, 11), 5, 3)
('1002', '자바 프로그래밍', '이몽룡', 25000, datetime.date(2021, 12, 12), 4, 1)
('1003', '인터넷프로그래밍', '성춘향', 30000, datetime.date(2019, 2, 10), 10, 2)
('1004', '안드로이드 프로그래밍', '변학도', 23000, datetime.date(2017, 10, 10), 2, 1)
('1005', '안드로이드 앱', '강길동', 26000, datetime.date(2020, 1, 11), 5, 2)
('1006', 'MS SQL SERVER 2014', '박지성', 35000, datetime.date(2020, 3, 25), 7, 3)
('1007', 'HTML & CSS', '손흥민', 18000, datetime.date(2023, 9, 30), 3, 1)
('1008', 'MFC 입문', '류현진', 20000, datetime.date(2015, 12, 12), 5, 1)
('1009', '안드로이드 게임 제작', '나길동', 33000, datetime.date(2018, 10, 31), 5, 2)
('1010', 'C++객체지향 프로그래밍', '추신수', 24000, datetime.date(2019, 4, 20), 2, 3)
('1011', 'JSP 웹 프로그래밍', '김연아', 27000, datetime.date(2020, 10, 23), 8, 1)
('1012', '해커들의 해킹 기법', '손연재', 32000, datetime.date(2018, 7, 7), 1, 2)
('1013', '자료구조', '홍길동', 19000, datetime.date(2020, 1, 20), 4, 1)
('1014', '파이썬', '성춘향', 35000, datetime.date(2023, 9, 25), 5, 2)
('1015', '자

In [23]:
# 도서번호, 도서명, 도서가격만 쿼리 실행 
sql = 'select bookNo, bookName, bookPrice from book';
cursor.execute(sql)
result = cursor.fetchall()
print(result)


(('1001', '데이터베이스 이론', 22000), ('1002', '자바 프로그래밍', 25000), ('1003', '인터넷프로그래밍', 30000), ('1004', '안드로이드 프로그래밍', 23000), ('1005', '안드로이드 앱', 26000), ('1006', 'MS SQL SERVER 2014', 35000), ('1007', 'HTML & CSS', 18000), ('1008', 'MFC 입문', 20000), ('1009', '안드로이드 게임 제작', 33000), ('1010', 'C++객체지향 프로그래밍', 24000), ('1011', 'JSP 웹 프로그래밍', 27000), ('1012', '해커들의 해킹 기법', 32000), ('1013', '자료구조', 19000), ('1014', '파이썬', 35000), ('1015', '자바스크립 & jQuery', 27000))


In [24]:
# 도서 가격이 30,000원 이상인 도서의 도서번호, 도서명, 가격 추출 
sql = 'select bookno,bookname,bookprice from book where bookprice >=30000'

cursor.execute(sql)
result = cursor.fetchall()
result


(('1003', '인터넷프로그래밍', 30000),
 ('1006', 'MS SQL SERVER 2014', 35000),
 ('1009', '안드로이드 게임 제작', 33000),
 ('1012', '해커들의 해킹 기법', 32000),
 ('1014', '파이썬', 35000))

# 다음과 같이 출력  
1003 인터넷프로그래밍 30000
1006 MS SQL SERVER 2014 35000
1009 안드로이드 게임 제작 33000
1012 해커들의 해킹 기법 32000
1014 파이썬 35000

In [25]:
for re in result:
    for r in re:
        print(r, end=" ")
    print()

1003 인터넷프로그래밍 30000 
1006 MS SQL SERVER 2014 35000 
1009 안드로이드 게임 제작 33000 
1012 해커들의 해킹 기법 32000 
1014 파이썬 35000 


In [ ]:
# 새로 시작하면
# (1) DB 연결
# (2) 커서 생성
# (3) 사용할 데이터베이스(스키마) 지정
# (4) select 구문 확인

In [44]:
# 날짜 출력 
sql = 'select bookDate from book'
cursor.execute(sql)
result = cursor.fetchall()
result 

((datetime.date(2019, 11, 11),),
 (datetime.date(2021, 12, 12),),
 (datetime.date(2019, 2, 10),),
 (datetime.date(2017, 10, 10),),
 (datetime.date(2020, 1, 11),),
 (datetime.date(2020, 3, 25),),
 (datetime.date(2023, 9, 30),),
 (datetime.date(2015, 12, 12),),
 (datetime.date(2018, 10, 31),),
 (datetime.date(2019, 4, 20),),
 (datetime.date(2020, 10, 23),),
 (datetime.date(2018, 7, 7),),
 (datetime.date(2020, 1, 20),),
 (datetime.date(2023, 9, 25),),
 (datetime.date(2019, 10, 22),),
 (datetime.date(2022, 11, 11),),
 (datetime.date(2022, 12, 10),))

In [46]:
# 각 튜플의 원소 출력하고
# datetime.date() 타입을 문자열로 변환 : strftime() 함수 사용
for i in range(len(result)):
    print(result[i][0].strftime('%Y년 %m월 %d일 %H시 %M분 %S초'))

2019년 11월 11일 00시 00분 00초
2021년 12월 12일 00시 00분 00초
2019년 02월 10일 00시 00분 00초
2017년 10월 10일 00시 00분 00초
2020년 01월 11일 00시 00분 00초
2020년 03월 25일 00시 00분 00초
2023년 09월 30일 00시 00분 00초
2015년 12월 12일 00시 00분 00초
2018년 10월 31일 00시 00분 00초
2019년 04월 20일 00시 00분 00초
2020년 10월 23일 00시 00분 00초
2018년 07월 07일 00시 00분 00초
2020년 01월 20일 00시 00분 00초
2023년 09월 25일 00시 00분 00초
2019년 10월 22일 00시 00분 00초
2022년 11월 11일 00시 00분 00초
2022년 12월 10일 00시 00분 00초


In [47]:
for i in range(len(result)):
    print(result[i][0].strftime('%Y-%m-%d'))

2019-11-11
2021-12-12
2019-02-10
2017-10-10
2020-01-11
2020-03-25
2023-09-30
2015-12-12
2018-10-31
2019-04-20
2020-10-23
2018-07-07
2020-01-20
2023-09-25
2019-10-22
2022-11-11
2022-12-10


In [30]:
# 연습문제
# 판매된 도서 중에서 '도서출판 강남'에서 출간된 도서에 대하여
# 고객명, 주문일, 도서명, 주문수량, 출판사명 출력
# 고객명 오름차순 정렬

In [51]:
sql = '''
         SELECT clientName, bsDate, bookName, bsQty, pubName
         FROM bookSale
             INNER JOIN book ON book.bookNo = bookSale.bookNo
             INNER JOIN publisher ON publisher.pubNo = book.pubNo
             INNER JOIN client ON client.clientNo = bookSale.clientNo
         WHERE pubName = '도서출판 강남'
         ORDER BY clientName
      ''';
cursor.execute(sql)
result = cursor.fetchall()
print(result)

(('베컴', datetime.date(2021, 7, 9), '인터넷프로그래밍', 2, '도서출판 강남'), ('옴바페', datetime.date(2022, 1, 2), '안드로이드 앱', 9, '도서출판 강남'), ('호날두', datetime.date(2020, 2, 20), '안드로이드 게임 제작', 5, '도서출판 강남'))


In [ ]:
########################################################################################
# INSERT

In [33]:
# 데이터 입력 받아서 book 테이블에 저장

In [40]:
bookNo = input('도서번호 입력 : ')
bookName = input('도서명 입력 : ')
bookAuthor = input('저자 입력 : ')
bookPrice = input('가격 입력 : ')
bookDate = input('출판일 입력 : ')
bookStock = input('재고 입력 : ')
pubNo = input('출판사번호 입력 : ')

도서번호 입력 :  1017
도서명 입력 :  db2
저자 입력 :  강길동
가격 입력 :  25000
출판일 입력 :  2022-12-10
재고 입력 :  7
출판사번호 입력 :  3


In [41]:
bookNo, bookName, bookAuthor, bookPrice, bookDate, bookStock, pubNo

('1017', 'db2', '강길동', '25000', '2022-12-10', '7', '3')

In [42]:
# INSERT 방법1
sql = "insert into book(bookNo, bookName, bookAuthor, bookPrice, bookDate, bookStock, pubNo) values('" \
      + bookNo + "', '" \
      + bookName + "','" \
      + bookAuthor + "'," \
      + bookPrice + ",'" \
      + bookDate + "'," \
      + bookStock + ",'" \
      + pubNo + "')"

cursor.execute(sql)

1

In [43]:
conn.commit() # 반드시 commit 해서 완료시켜야 DB에 반영됨

In [52]:
# INSERT 방법2
bookNo = input('도서번호 입력 : ')
bookName = input('도서명 입력 : ')
bookAuthor = input('저자 입력 : ')
bookPrice = input('가격 입력 : ')
bookDate = input('출판일 입력 : ')
bookStock = input('재고 입력 : ')
pubNo = input('출판사번호 입력 : ')

도서번호 입력 :  1018
도서명 입력 :  sql
저자 입력 :  이길동
가격 입력 :  30000
출판일 입력 :  2023-09-27
재고 입력 :  1
출판사번호 입력 :  1


In [53]:
# 모든 열에 데이터 입력 시 열이름 리스트 생략 가능
sql = "insert into book values(%s, %s, %s, %s, %s, %s, %s)" # 위치와 데이터 타입 지정해 놓고
values = (bookNo, bookName, bookAuthor, bookPrice, bookDate, bookStock, pubNo) # 실제 전달될 값
cursor.execute(sql, values) # 값도 같이 전달
conn.commit()

In [54]:
sql = 'select * from book'
cursor.execute(sql)
result = cursor.fetchall()
result

(('1001', '데이터베이스 이론', '홍길동', 22000, datetime.date(2019, 11, 11), 5, 3),
 ('1002', '자바 프로그래밍', '이몽룡', 25000, datetime.date(2021, 12, 12), 4, 1),
 ('1003', '인터넷프로그래밍', '성춘향', 30000, datetime.date(2019, 2, 10), 10, 2),
 ('1004', '안드로이드 프로그래밍', '변학도', 23000, datetime.date(2017, 10, 10), 2, 1),
 ('1005', '안드로이드 앱', '강길동', 26000, datetime.date(2020, 1, 11), 5, 2),
 ('1006',
  'MS SQL SERVER 2014',
  '박지성',
  35000,
  datetime.date(2020, 3, 25),
  7,
  3),
 ('1007', 'HTML & CSS', '손흥민', 18000, datetime.date(2023, 9, 30), 3, 1),
 ('1008', 'MFC 입문', '류현진', 20000, datetime.date(2015, 12, 12), 5, 1),
 ('1009', '안드로이드 게임 제작', '나길동', 33000, datetime.date(2018, 10, 31), 5, 2),
 ('1010', 'C++객체지향 프로그래밍', '추신수', 24000, datetime.date(2019, 4, 20), 2, 3),
 ('1011', 'JSP 웹 프로그래밍', '김연아', 27000, datetime.date(2020, 10, 23), 8, 1),
 ('1012', '해커들의 해킹 기법', '손연재', 32000, datetime.date(2018, 7, 7), 1, 2),
 ('1013', '자료구조', '홍길동', 19000, datetime.date(2020, 1, 20), 4, 1),
 ('1014', '파이썬', '성춘향', 35000, dateti

In [ ]:
##########################################################################################

In [ ]:
# UPDATE
# 주의! : commit 

In [ ]:
# 고객명이 '메시'인 고객의 전화번호 값 수정 : 010-1111-1111을 010-1234-1234로 변경

In [56]:
sql = '''
         UPDATE client
         SET clientPhone = '010-1234-1234'
         WHERE clientName = '메시'
      ''';
cursor.execute(sql)
conn.commit()

In [57]:
sql = 'select * from client'
cursor.execute(sql)
result = cursor.fetchall()
result

(('1', '메시', '010-1234-1234', '서울시', datetime.date(1987, 6, 24), '축구', '남'),
 ('2', '호날두', '010-2222-2222', '천안', datetime.date(1984, 2, 5), None, '남'),
 ('3', '샤라포바', '010-3333-3333', '제주', datetime.date(1987, 4, 19), None, '여'),
 ('4', '베컴', '010-4444-4444', '안양', datetime.date(1994, 5, 2), '영화', '남'),
 ('5', '네이마르', '010-5555-5555', '수원', datetime.date(1992, 2, 5), '게임', '남'),
 ('6', '윌리엄스', '010-6666-6666', '서울', datetime.date(1981, 9, 26), '테니스', '여'),
 ('7', '옴바페', '010-7777-7777', '천안', datetime.date(1991, 1, 7), ' ', '남'),
 ('8', '살라', '010-8888-8888', '부산', datetime.date(1982, 4, 22), '', '남'))

In [59]:
# 입력된 값으로 데이터 수정
# 수정하고자 하는 기본키 값을 받고
# 수정할 데이터 입력
bookNo = input('수정할 도서번호 입력 : ')
bookName = input('도서명 입력 : ')
bookAuthor = input('저자 입력 : ')
bookPrice = input('가격 입력 : ')
bookDate = input('출판일 입력 : ')
bookStock = input('재고 입력 : ')
pubNo = input('출판사번호 입력 : ')

# '1001', '데이터베이스 이론', '홍길동', 22000, datetime.date(2019, 11, 11), 5, 3
# '데이터베이스 이론 및 실기', 25000, 재고 7

수정할 도서번호 입력 :  1001
도서명 입력 :  데이터베이스 이론 및 실기
저자 입력 :  홍길동
가격 입력 :  25000
출판일 입력 :  2019-11-11
재고 입력 :  7
출판사번호 입력 :  3


In [ ]:
# update 테이블 set 열=값, 열=값, .... where 열 = 값

In [62]:
sql = "UPDATE book SET bookName=%s, bookAuthor=%s, " \
      + "bookPrice=%s, bookDate=%s, bookStock=%s, pubNo=%s WHERE bookNo=%s"

values = (bookName, bookAuthor, bookPrice, bookDate, bookStock, pubNo, bookNo) # 실제 전달될 값
cursor.execute(sql, values) # 값도 같이 전달
conn.commit()

In [69]:
sql = 'select * from book'
cursor.execute(sql)
result = cursor.fetchall()
result

(('1001', '데이터베이스 이론 및 실기', '홍길동', 25000, datetime.date(2019, 11, 11), 7, 3),
 ('1002', '자바 프로그래밍', '이몽룡', 25000, datetime.date(2021, 12, 12), 4, 1),
 ('1003', '인터넷프로그래밍', '성춘향', 30000, datetime.date(2019, 2, 10), 10, 2),
 ('1004', '안드로이드 프로그래밍', '변학도', 23000, datetime.date(2017, 10, 10), 2, 1),
 ('1005', '안드로이드 앱', '강길동', 26000, datetime.date(2020, 1, 11), 5, 2),
 ('1006',
  'MS SQL SERVER 2014',
  '박지성',
  35000,
  datetime.date(2020, 3, 25),
  7,
  3),
 ('1007', 'HTML & CSS', '손흥민', 18000, datetime.date(2023, 9, 30), 3, 1),
 ('1008', 'MFC 입문', '류현진', 20000, datetime.date(2015, 12, 12), 5, 1),
 ('1009', '안드로이드 게임 제작', '나길동', 33000, datetime.date(2018, 10, 31), 5, 2),
 ('1010', 'C++객체지향 프로그래밍', '추신수', 24000, datetime.date(2019, 4, 20), 2, 3),
 ('1011', 'JSP 웹 프로그래밍', '김연아', 27000, datetime.date(2020, 10, 23), 8, 1),
 ('1012', '해커들의 해킹 기법', '손연재', 32000, datetime.date(2018, 7, 7), 1, 2),
 ('1013', '자료구조', '홍길동', 19000, datetime.date(2020, 1, 20), 4, 1),
 ('1014', '파이썬', '성춘향', 35000, d

In [ ]:
#############################################################################################

In [64]:
# DELETE
# 주의! : commit

In [65]:
sql = 'delete from book where bookNo = "1018"'
cursor.execute(sql)
conn.commit()

In [67]:
# 삭제할 도서번호 입력 받아서 삭제
bookNo = input('삭제할 도서번호 입력 : ')

삭제할 도서번호 입력 :  1017


In [68]:
sql = "DELETE FROM book WHERE bookNo=%s"

values = (bookNo) # 실제 전달될 값
cursor.execute(sql, values) # 값도 같이 전달
conn.commit()

In [ ]:
#############################################################################################

In [70]:
# 저장 프로시저 호출

In [71]:
# (1) 매개변수 없는 경우
# 한번이라도 주문한 적이 있는 모든 고객의 고객등급 출력
# 고객번호, 고객명, 총주문액, 고객등급 출력
# 프로시저명 : clientAllRatingProc
cursor.callproc('clientAllRatingProc')
result = cursor.fetchall()
result

(('2', '호날두', Decimal('265000'), '최우수'),
 ('4', '베컴', Decimal('60000'), '보통'),
 ('8', '살라', Decimal('23000'), '관심고객'),
 ('7', '옴바페', Decimal('257000'), '최우수'),
 ('3', '샤라포바', Decimal('70000'), '보통'),
 ('1', '메시', Decimal('108000'), '우수'))

In [72]:
# 매개변수가 있는 경우 : IN 매개변수만 있는 경우
# 방법 1
# 고객의 연령대를 출력하는 프로시저
# 고객번호를 전달 받아서, 해당 고객의 생년월일 확인 후 '~'년대생' 출력
# 예 : 1990년대생, 1980년대생, 1970년대생 출력
# 고객번호, 고객명, 연령대
# 프로시저명 : clientAgeCheckProc;
# 호출하면서 값 전달
# CALL clientAgeCheckProc('1');

In [73]:
# 방법1
cursor.callproc('clientAgeCheckProc', {'1'})
result = cursor.fetchall()
result

(('1', '메시', datetime.date(1987, 6, 24), '1980년대생'),)

In [74]:
# 방법2
cursor.callproc('clientAgeCheckProc', ('2'))
result = cursor.fetchall()
result

(('2', '호날두', datetime.date(1984, 2, 5), '1980년대생'),)

In [75]:
# 방법3
cursor.callproc('clientAgeCheckProc', '3')
result = cursor.fetchall()
result

(('3', '샤라포바', datetime.date(1987, 4, 19), '1980년대생'),)

In [76]:
# 방법4
sql = "CALL clientAgeCheckProc('4')"
cursor.execute(sql)
result = cursor.fetchall()
result

(('4', '베컴', datetime.date(1994, 5, 2), '1990년대생'),)

In [81]:
# 매개변수가 있는 경우 : IN / OUT 매개변수만 있는 경우
# 도서가격을 전달하면, 전달된 도서가격 이상인 도서 수를 반환
# 프로시저명 : bookPriceCountProc;
# 프로시저 호출
# CALL bookPriceCountProc(30000, @bookCount);
# SELECT @bookCount;

sql = "CALL bookPriceCountProc(25000, @bookCount)"
cursor.execute(sql)
sql = "SELECT @bookCount"
cursor.execute(sql)

# result = cursor.fetchall()
# result

# print(result[0][0])

result = cursor.fetchone() # 1개 값이 반환되는 경우
result
print(result[0])

10


In [82]:
cursor.close()
conn.close()